### PR TITLE
feat: add lazyLoad option at enforcer init method

### PR DESCRIPTION
**Pull request by Gabriel-403**
Fix: #168

### CODE REVIEW

File: src/enforcer.ts [hunk 1]
```diff
@@ -26,30 +26,33 @@ export class Enforcer extends ManagementEnforcer {
...
-  public async initWithFile(modelPath: string, policyPath: string): Promise<void> {
+  public async initWithFile(modelPath: string, policyPath: string, lazyLoad: boolean =  false): Promise<void> {
     const a = new FileAdapter(policyPath);
-    await this.initWithAdapter(modelPath, a);
+    await this.initWithAdapter(modelPath, a, lazyLoad);
   }
 
   /**
-   * initWithFile initializes an enforcer with a model file and a policy file.
+   * initWithFile initializes an eforcer with a model file and a policy file.
```
Comment (hsluoyz) [inline src/enforcer.ts:37]: no change

File: src/enforcer.ts [hunk 2]
```diff
@@ -26,30 +26,33 @@ export class Enforcer extends ManagementEnforcer {
...
    * @param modelPath model file path
    * @param policyString policy CSV string
+   * @param lazyLoad whether load policy when initial
```
Comment (hsluoyz) [inline src/enforcer.ts:40]: whether to load policy at initial time

File: src/enforcer.ts [hunk 3]
```diff
@@ -58,16 +61,17 @@ export class Enforcer extends ManagementEnforcer {
    * initWithModelAndAdapter initializes an enforcer with a model and a database adapter.
    * @param m model instance
    * @param adapter current adapter instance
+   * @param lazyLoad whether load policy when initial
```
Comment (hsluoyz) [inline src/enforcer.ts:64]: whether to load policy at initial time

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@Gabriel-403 plz fix

image

File: src/enforcer.ts [hunk 4]
```diff
@@ -26,30 +26,33 @@ export class Enforcer extends ManagementEnforcer {
    * initWithFile initializes an enforcer with a model file and a policy file.
    * @param modelPath model file path
    * @param policyPath policy file path
+   * @param lazyLoad lazyLoad whether to load policy at initial time
```
Comment (hsluoyz) [inline src/enforcer.ts:29]: two lazyLoad lazyLoad

File: src/enforcer.ts [hunk 5]
```diff
@@ -26,30 +26,33 @@ export class Enforcer extends ManagementEnforcer {
...
    */
-  public async initWithFile(modelPath: string, policyPath: string): Promise<void> {
+  public async initWithFile(modelPath: string, policyPath: string, lazyLoad = false): Promise<void> {
```
Comment (hsluoyz) [inline src/enforcer.ts:31]: Why no param type like boolean? Is this OK for TS?

File: src/enforcer.ts [hunk 6]
```diff
@@ -26,30 +26,33 @@ export class Enforcer extends ManagementEnforcer {
...
   }
 
   /**
    * initWithFile initializes an enforcer with a model file and a policy file.
    * @param modelPath model file path
    * @param policyString policy CSV string
+   * @param lazyLoad whether to load policy at initial time
    */
-  public async initWithString(modelPath: string, policyString: string): Promise<void> {
+  public async initWithString(modelPath: string, policyString: string, lazyLoad = false): Promise<void> {
```
Comment (hsluoyz) [inline src/enforcer.ts:42]: Why no param type like boolean? Is this OK for TS?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@nodece @Zxilly plz review.

File: src/enforcer.ts (hunk 5 above)
Comment (nodece) [reply src/enforcer.ts:31]: no problem.

**Review by Zxilly — APPROVED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**